src/il_representations/il/ppo_gail.py

```diff
@@ -108,14 +105,88 @@ def update(self, rollouts):
                                          self.max_grad_norm)
                 self.optimizer.step()
 
-                value_loss_epoch += stats_dict['value_loss']
-                action_loss_epoch += stats_dict['action_loss']
-                dist_entropy_epoch += stats_dict['dist_entropy']
-
-        num_updates = self.ppo_epoch * self.num_mini_batch
 
-        value_loss_epoch /= num_updates
-        action_loss_epoch /= num_updates
-        dist_entropy_epoch /= num_updates
+class DiscriminatorDataset(IterableDataset):
+    """Dataset for training discriminator. Capable of mixing both expert and
+    novice datasets, and of adding more novice samples at execution time.
+
+    Uses label convention that 1 = real and 0 = fake (so discriminator
+    estimates p(fake|data))."""
+    def __init__(self, expert_dataset, novice_circ_buffer_size=4096):
+        pass
+
+
+class Discriminator:
+    """Implements only the GAIL discriminator."""
+    def __init__(
+        self,
+        *,
+        discriminator,
+    ):
+        self.discriminator = discriminator
+
+    def all_trainable_params(self):
+        """Trainable parameters for discriminator."""
+        return self.discriminator.parameters()
+
+    def make_data_iter(self, expert_dataset, novice_dataset, augmentation_fn,
```

Comment:
It seems the `novice_dataset` is never used here?

Reply:
This code is currently broken, along with the rest of the joint training GAIL code. I'm going to delete that junk entirely, since I probably won't need it (and can rescue from commit history if it turns out that I do need it).